source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -240,7 +246,25 @@ absl::Status ProcessorState::handleBodyResponse(const BodyResponse& response) {
       auto queued_chunk = dequeueStreamingChunk(false);
       ENVOY_BUG(queued_chunk, "Bad partial body callback state");
       auto chunk = std::move(*queued_chunk);
+      if (common_response.has_header_mutation()) {
+        if (headers_ != nullptr) {
+          ENVOY_LOG(debug, "Applying header mutations to buffered body message");
+          const auto mut_status = MutationUtils::applyHeaderMutations(
+              common_response.header_mutation(), *headers_,
+              common_response.status() == CommonResponse::CONTINUE_AND_REPLACE,
+              filter_.config().mutationChecker(), filter_.stats().rejected_header_mutations_);
+          if (!mut_status.ok()) {
+            return mut_status;
+          }
+        } else {
+          ENVOY_LOG(debug, "Response had header mutations but headers aren't available");
+        }
+      }
       if (common_response.has_body_mutation()) {
+        if (headers_ != nullptr) {
+          // Always reset the content length here to prevent later problems.
+          headers_->removeContentLength();
```

Comment:
Is this code block the same as the one above that handles evaluating the header mutation? If so, is there a way to resuse the code to make it obivous that we're doing the same thing?

Reply:
Well, there are three places where we call applyHeaderMutations and two where we call applyBodyMutations, but each time the context or the parameters are different -- I don't see a further opportunity to "DRY up" that code from where it already is. If it helps we can add more comments, or adjust the log messages to make them more consistent, if you think that helps. Otherwise, if I'm missing the point, I'm happy to look for more opportunities.

Reply:
I'm going to resolve this and the other comments, but please re-open if you have more comments!